java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java

```diff
@@ -686,6 +693,38 @@ public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
                 }
                 return CompletableFuture.completedFuture(true);
             }
+            case Server.NBLS_DOCUMENT_SYMBOLS: {
+                List<DocumentSymbol> result = new ArrayList<>();
+                try {
+                    List<Object> arguments = params.getArguments();
+                    String source = ((JsonPrimitive) arguments.get(0)).getAsString();
+                    String query = arguments.size() > 1 ? ((JsonPrimitive)arguments.get(1)).getAsString() : "";
+                    FileObject file = Utils.fromUri(source);
+                    Document rawDoc = server.getOpenedDocuments().getDocument(source);
```

Comment:
Should the query work just for opened (didOpen received) documents ?

Reply:
Yes, the existing alternative `nbls.workspace.symbols` that returns symbols found in all project's sources is too heavyweight/slow for work with a single document opened in editor....